src/main/java/duke/util/TaskList.java

```diff
@@ -0,0 +1,106 @@
+package duke.util;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+public class TaskList {
+    private ArrayList<Task> arr;
+
+    /**
+     * Constructor of class TaskList.
+     * @param arr The input ArrayList that stores the past tasks that were stored
+     *            in a file if there were any.
+     */
+    public TaskList(ArrayList<Task> arr) {
+        // arr is given from the duke.util.Storage.load()
+        this.arr = arr;
+    }
+
+    /**
+     * To find tasks containing the String keyword.
+     * @param keyword The input keyword.
+     * @return An ArrayList containing the tasks containing the keyword.
+     */
+    public ArrayList<Task> find(String keyword) {
+        ArrayList<Task> containsKey = new ArrayList<>();
+        for (Task task : this.arr) {
+            if (task.toString().contains(keyword)) {
+                containsKey.add(task);
+            }
+        }
+        return containsKey;
+    }
+
+    /**
+     * To add a Task into the TaskList.
+     * @param instance The task added into the TaskList.
+     */
+
+    public void add(Task instance) {
+        this.arr.add(instance);
+    }
+
+    /**
+     * To delete a task from the TaskList.
+     * @param num The index number of the Task in the TaskList.
+     * @return The Task that got deleted.
+     */
+    public Task delete(int num) {
```

Comment:
would it be better to let the user know that they have entered an invalid index instead of returning null? or was this handled somewhere else